lib/util/packed_xml.js

```diff
@@ -0,0 +1,173 @@
+            }
+        }
+
+        res.newFormat = true;
+        res.bRand = data[dataLen - 3];
+        res.nVer = data[dataLen - 2] ^ data[dataLen - 25];
+
+        return res;
+
+    };
+
+    this.unobfuscate = function(packedXmlData) {
+
+        var dataLen, unpackedSize, fileInfo;
+
+        dataLen = packedXmlData.length;
+
+        // check header
+        if (packedXmlData.slice(0, 10).toString() !== "PACKED_XML") {
+            throw "This is not PACKED_XML";
+        }
+
+        // get unpacked data size
+        unpackedSize = packedXmlData.readUInt32LE(11);
+
+        // get lzma obfuscation version
+        fileInfo = this.getVersion(packedXmlData.slice(15), dataLen - 15);
+        if (fileInfo.newFormat) {
+            if (fileInfo.nVer === 1) {
+                // fixing header
+                var count1 = (dataLen - 28) > 1024 ? 512 : (dataLen - 28) / 2;
+                for (var count = 0; count < count1; count++) {
+                    packedXmlData[15 + count] ^= packedXmlData[15 + dataLen - 28 - count1 + count] ^ fileInfo.bRand;
+                }
+            } else {
+                throw "Unsupported PACKED_XML version: " + fileInfo.nVer;
+            }
+        } else {
+            // xor first 160 bytes after header
+            dataLen - 15 > 160 ? count1 = 160 : count1 = dataLen - 15;
+            for (var count = 0; count < count1; count++) {
+                packedXmlData[15 + count] ^= xorStr[count % 16];
+            }
+        }
+
+        // cut header
+        return packedXmlData.slice(16);
+
+    };
+
+    this.unlzma = function(data) {
+
+        var lzmaData, decoder, outSize, inStream, outStream;
+
+        // try to unobfuscate LZMA data
+        lzmaData = this.unobfuscate(data);
+        if (lzmaData === undefined) {
+            throw "Error unobfuscating data";
+        }
+
+        // construct input stream for LZMA decoder
+        inStream = new Stream.inStream(lzmaData);
+
+        // construct decoder and set properties from stream
+        var decoder = new lzmaDecoder.Decoder();
+        if (!decoder.setDecoderProperties(inStream)) {
+            throw "Incorrect LZMA stream properties";
+        }
+
+        // data size from LZMA header
+        outSize = inStream.readUInt32LE();
+
+        // construct output stream and reserve space
+        outStream = new Stream.outStream(outSize);
+
+        // skip 4 zeroes...
+        inStream.readUInt32LE();
+
+        // actually decode
+        if (!decoder.decode(inStream, outStream, outSize)) {
+            throw "Error in LZMA data stream";
+        };
+
+        return outStream.data;
+    };
+
+    this.unpack = function(data) {
+
+        var xmlBuf, xml, el, partName;
+        var res = {};
+
+        xmlBuf = this.unlzma(data);
+
+        if (xmlBuf.slice(0, 5).toString() !== "<?xml") {
+            throw Error("This is not XML");
+        }
+
+        xml = new xmldoc.XmlDocument(xmlBuf.toString());
+        xmlBuf = undefined;

```

Comment:
I can go to regexp or string.indexof... But parser I used is not such a big thing. Is that a problem?